lib/filters/http-headers.js

```diff
@@ -20,6 +20,11 @@ function httpHeaders (obj) {
 
 function filterSensitiveHeaders (headers) {
   for (const key in headers) {
```

Comment:
Unrelated to this change: Wouldn't it typically be faster to iterate through the set of sensitive headers, rather than iterating through the set of given headers? I say "typically" because my guess is request and response headers generally number greater than 3. Anyway, if you agree, I can open a ticket to consider changing this. Low prio, this is possible premature optimization.

Reply:
This is good optimization to consider, but I agree it may be premature (especially given the unexpected gymnastic to conform to the spec and maintain 3.x compatibility).  I've opened an issue to consider this optimization here: https://github.com/elastic/apm-agent-nodejs/issues/1923.